api/types/audit.go

```diff
@@ -0,0 +1,258 @@
+}
+
+// Type gets the audit backend type.
+func (c *ClusterAuditConfigV2) Type() string {
+	return c.Spec.Type
+}
+
+// SetType sets the audit backend type.
+func (c *ClusterAuditConfigV2) SetType(backendType string) {
+	c.Spec.Type = backendType
+}
+
+// Region gets a cloud provider region.
+func (c *ClusterAuditConfigV2) Region() string {
+	return c.Spec.Region
+}
+
+// SetRegion sets a cloud provider region.
+func (c *ClusterAuditConfigV2) SetRegion(region string) {
+	c.Spec.Region = region
+}
+
+// ShouldUploadSessions returns whether audit config
+// instructs server to upload sessions.
+func (c *ClusterAuditConfigV2) ShouldUploadSessions() bool {
+	return c.Spec.AuditSessionsURI != ""
+}
+
+// AuditSessionsURI gets the audit sessions URI.
+func (c *ClusterAuditConfigV2) AuditSessionsURI() string {
+	return c.Spec.AuditSessionsURI
+}
+
+// SetAuditSessionsURI sets the audit sessions URI.
+func (c *ClusterAuditConfigV2) SetAuditSessionsURI(uri string) {
+	c.Spec.AuditSessionsURI = uri
+}
+
+// AuditEventsURIs gets the audit events URIs.
+func (c *ClusterAuditConfigV2) AuditEventsURIs() []string {
+	return c.Spec.AuditEventsURI
+}
+
+// SetAuditEventsURIs sets the audit events URIs.
+func (c *ClusterAuditConfigV2) SetAuditEventsURIs(uris []string) {
+	c.Spec.AuditEventsURI = uris
+}
+
+// EnableContinuousBackups is used to enable (or disable) PITR (Point-In-Time Recovery).
+func (c *ClusterAuditConfigV2) EnableContinuousBackups() bool {
+	return c.Spec.EnableContinuousBackups
+}
+
+// EnableAutoScaling is used to enable (or disable) auto scaling policy.
+func (c *ClusterAuditConfigV2) EnableAutoScaling() bool {
+	return c.Spec.EnableAutoScaling
+}
+
+// ReadMaxCapacity is the maximum provisioned read capacity.
+func (c *ClusterAuditConfigV2) ReadMaxCapacity() int64 {
+	return c.Spec.ReadMaxCapacity
+}
+
+// ReadMinCapacity is the minimum provisioned read capacity.
+func (c *ClusterAuditConfigV2) ReadMinCapacity() int64 {
+	return c.Spec.ReadMinCapacity
+}
+
+// ReadTargetValue is the ratio of consumed read to provisioned capacity.
+func (c *ClusterAuditConfigV2) ReadTargetValue() float64 {
+	return c.Spec.ReadTargetValue
+}
+
+// WriteMaxCapacity is the maximum provisioned write capacity.
+func (c *ClusterAuditConfigV2) WriteMaxCapacity() int64 {
+	return c.Spec.WriteMaxCapacity
+}
+
+// WriteMinCapacity is the minimum provisioned write capacity.
+func (c *ClusterAuditConfigV2) WriteMinCapacity() int64 {
+	return c.Spec.WriteMinCapacity
+}
+
+// WriteTargetValue is the ratio of consumed write to provisioned capacity.
+func (c *ClusterAuditConfigV2) WriteTargetValue() float64 {
+	return c.Spec.WriteTargetValue
+}
+
+// CheckAndSetDefaults verifies the constraints for ClusterAuditConfig.
+func (c *ClusterAuditConfigV2) CheckAndSetDefaults() error {
+	// Make sure we have defaults for all metadata fields.
+	err := c.Metadata.CheckAndSetDefaults()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if c.Version == "" {
+		c.Version = V2
+	}
```

Comment:
let's use the `setStaticFields` pattern from https://github.com/gravitational/teleport/pull/6846 here too